sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -172,6 +252,55 @@ def on_exception(self, request):
         return
 
 
+class MultitenantCredentialPolicy(BearerTokenCredentialPolicy):
+    """Adds a bearer token Authorization header to requests, for the tenant provided in authentication challenges.
+
+    See https://docs.microsoft.com/azure/active-directory/develop/claims-challenge for documentation on AAD
+    authentication challenges.
+
+    :param credential: The credential.
+    :type credential: ~azure.core.TokenCredential
+    :param str scopes: Lets you specify the type of access needed.
+    :keyword bool enable_tenant_discovery: Determines if tenant discovery should be enabled. Defaults to True.
+    :keyword bool enable_scopes_discovery: Determines if scopes from authentication challenges should be provided to
+        token requests, instead of the scopes given to the policy's constructor. Defaults to True.
+    :raises: :class:`~azure.core.exceptions.ServiceRequestError`
+    """
+
+    def __init__(self, credential: "TokenCredential", *scopes: str, **kwargs: "Any") -> None:
+        self._discover_tenant = kwargs.pop("enable_tenant_discovery", True)
+        self._discover_scopes = kwargs.pop("enable_scopes_discovery", True)
```

Comment:
Have we used the `enable_` prefix elsewhere in credentials/identity?
We could potentially use something shorter like `discover_scopes`/`discover_tenant`. But if we've used `enable_` elsewhere we can stick with the pattern.

Reply:
Why `kwargs.pop()` with default value and not keyword-only arguments?